src/errors.rs

```diff
@@ -1,13 +1,15 @@
 use std::env;
 use std::error;
 use std::fmt;
+use std::fmt::Display;
 use std::io;
 
 #[derive(Debug)]
 #[non_exhaustive]
```

Comment:
Thanks for this! I did not know about `non_exhaustive` until this change and it is what allowed this PRs conservative approach of not breaking existing users / requiring a major semver bump.